pootle/apps/pootle_store/views.py

```diff
@@ -452,7 +453,10 @@ def get_units(request):
     if 'virtualfolder' in settings.INSTALLED_APPS:
         from virtualfolder.helpers import extract_vfolder_from_path
 
-        vfolder, pootle_path = extract_vfolder_from_path(pootle_path)
+        vfolder, pootle_path = extract_vfolder_from_path(
+            pootle_path,
+            vfti=VirtualFolderTreeItem.objects.select_related(
+                "directory", "vfolder"))
```

Comment:
well im thinking that we should make this function as extensible as possible - even though my tests seem to suggest that in these 2 circumstances the optimization is the same.